CMSIS/NN/Include/arm_nnfunctions.h

```diff
@@ -445,6 +445,65 @@ extern    "C"
                                                       q15_t * bufferA,
                                                       q7_t * bufferB);
 
+  /**
+   * @brief Fast s8 version for 1x1 convolution (non-square shape)
+   * @param[in]      input         pointer to input tensor.  Format: [H, W, in_ch]
+   * @param[in]      input_x       input tensor dimension x
+   * @param[in]      input_y       input tensor dimension y
+   * @param[in]      input_ch      number of input tensor channels
+   * @param[in]      kernel        pointer to kernel weights. Format: [out_ch, H, W, in_ch]
+   * @param[in]      output_ch     number of filters, i.e., output tensor channels
+   * @param[in]      pad_x         padding size x
+   * @param[in]      pad_y         padding size y
+   * @param[in]      stride_x      convolution stride x
+   * @param[in]      stride_y      convolution stride y
+   * @param[in]      bias          pointer to bias
+   * @param[in,out]  output        pointer to output tensor.  Format: [H, W, out_ch]
+   * @param[in]      output_shift  pointer to per output channel requantization shift parameter.
+   * @param[in]      output_mult   pointer to per output channel requantization multiplier parameter.
+   * @param[in]      out_offset    output tensor offset. Range: int8
+   * @param[in]      input_offset input tensor offset. Range: int8
+   * @param[in]      out_activation_min   Minimum value to clamp the output to. Range: int8
+   * @param[in]      out_activation_max   Minimum value to clamp the output to. Range: int8
+   * @param[in]      output_x  output tensor width
+   * @param[in]      output_y  output tensor height
+   * @param[in]      buffer_a  pointer to buffer space used for input optimization(partial im2col) and is necessary
+   *                           when ARM_MATH_LOOPUNROLL and ARM_MATH_DSP is defined.
+   *                           Required space: 2 * input_ch * sizeof(q15_t) bytes
+   * @return     The function returns either
+   *                  <code>ARM_MATH_SIZE_MISMATCH</code> if argument constraints fail. or,
+   *                  <code>ARM_MATH_SUCCESS</code> on successful completion.
+   *
+   * @details
+   *   - Supported framework : TensorFlow Lite Micro
+   *   - The following constrains on the arguments apply
+   *      -# output_ch and input_ch are multiples of 2
+   *      -# padding equals 0
+   *      -# Stride equals 1
```

Comment:
If pad and stride must be 0 and 1 respectively, what have the parameters in the interface?

Reply:
@freddan80  If we start with kernel_x and kernel_y parameters, it is a function intended for  1x1 convolution, and as the name of function and file has it, it is omitted from the argument list . i.e, quite obvious and is the base functionality of this function.

But, the assumptions/requirements for padding and stride values are at a level lower which I am not sure is always the case(0 and 1) when using a 1x1 convolution. So, for that minor reason and mainly to avoid a case where it is deployed and is accidentally/silently used for non compliant stride and pad values, it is included in the argument list.

Reply:
Rebased

Reply:
ok, fair enough